### ISSUE TITLE
container exector can't execute pre.sh

### ISSUE THREAD

**Issue by HenryGuo1019** [kubeshop/testkube#4719]
### The ERROR
/data/entrypoint.sh: 2: prerun.sh: not found

### The container exector:
apiVersion: executor.testkube.io/v1
kind: Executor
metadata:
  name: scrap-kubeconfig-executor
  namespace: testkube
spec:
  image: ecr-sh.yun.qianxin-inc.cn/alkaid/testkube-tools:v0.0.1
  command:
    - bash
  executor_type: container
  types:
    - scrap-kubeconfig/test

### The pre.sh
sshpass -p "$password" scp $user@$ip:/root/.kube/config ./

### The definition of test
apiVersion: tests.testkube.io/v3
kind: Test
metadata:
  name: init-kubeconfig-test
  namespace: testkube
  labels:
    executor: scrap-kubeconfig-executor
    test-type: scrap-kubeconfig-test
spec:
  type: scrap-kubeconfig/test
  executionRequest:
    preRunScript: "sshpass -p "$password" scp $user@$ip:/root/.kube/config ./"
    executePostRunScriptBeforeScraping: false

**Comment by HILLJAY1019** [comment]
> thank you @HILLJAY1019 we will check it out

You'are welcome , i think it should be like the other exector which exectute the pre.sh in a /tmp dir like /tmp/runscript1679809349.sh. For now ,i only see this bug in the container exector,the other exector like k6 is fine.

**Comment by windowsrefund** [comment]
Seeing the same thing when working with a container executor.

**Comment by rangoo94** [comment]
Hi there!

* @HenryGuo1019, do you still have this problem on the latest version?
* @windowsrefund, could you share your failing definition of Test and Executor?

I tried to reproduce the problem on the latest Testkube (`v1.16.18`), but it worked for me well.

I built example similar to the one from the issue:
```yaml
apiVersion: executor.testkube.io/v1
kind: Executor
metadata:
  name: prerun-test-executor
  namespace: testkube
spec:
  image: mcr.microsoft.com/playwright:v1.32.3-focal
  command:
    - "bash"
  executor_type: container
  types:
    - prerun-test-executor/test
---
apiVersion: tests.testkube.io/v3
kind: Test
metadata:
  name: prerun-test
  namespace: testkube
spec:
  type: prerun-test-executor/test
  content:
    type: git
    repository:
      type: git
      uri: https://github.com/kubeshop/testkube
      branch: develop
      path: test/playwright/executor-tests/playwright-project
      workingDir: test/playwright/executor-tests/playwright-project
  executionRequest:
    preRunScript: "npm ci"
    args:
      - "-c"
      - "npx --yes playwright@1.32.3 test"
    executePostRunScriptBeforeScraping: false
```

**Comment by rangoo94** [comment]
Hi there! Sorry for the inconvenience.

Looks like the `preRunScript` is failing when the Test has no `content`, or the `content` is `type: string`. We will take a look at it ASAP.

**Comment by rangoo94** [comment]
Minimum reproducible example:
```yaml
apiVersion: executor.testkube.io/v1
kind: Executor
metadata:
  name: prerun-test-executor
  namespace: testkube
spec:
  executor_type: container
  image: bash:latest
  types:
    - prerun-test-executor/test
---
apiVersion: tests.testkube.io/v3
kind: Test
metadata:
  name: prerun-test
  namespace: testkube
spec:
  type: prerun-test-executor/test
  executionRequest:
    preRunScript: "echo 'abc'"
```

**Comment by rangoo94** [comment]
@windowsrefund, @HenryGuo1019,

After digging deeper, the issue has been introduced in [**testkube#4489**](https://github.com/kubeshop/testkube/pull/4489) PR.

The pre-run and post-run script are put in the `workingDir`, which defaults to the image one. It often defaults to `/`. We have the volume attached at `/data` only, so it's built in the init container, but not available in the actual container.

We will try to release the fix today though. As a temporary workaround, I could suggest adding `content.repository` with `workingDir: /data`.

**Comment by rangoo94** [comment]
@windowsrefund, @HenryGuo1019,

The fix has been released in `v1.16.19` (Helm chart: `1.16.24`). 🎉 You may run `testkube upgrade` command to upgrade your Testkube server.

Sorry that it took so long for us, I hope that you like Testkube anyway 🙂

**Comment by windowsrefund** [comment]
Seems this has been fixed in 1.16.19 which means this can be closed.
